salt/minion.py

```diff
@@ -1365,6 +1366,15 @@ def _handle_decoded_payload(self, data):
                 self.functions, self.returners, self.function_errors, self.executors = self._load_modules()
                 self.schedule.functions = self.functions
                 self.schedule.returners = self.returners
+
+        process_count_max = self.opts.get('process_count_max')
```

Comment:
I have some real concerns here. This is perhaps the most performance-sensitive area of the minion and I really don't like introducing more disk hits and de-serialization overhead.

I would feel much more comfortable if this configuration option were disabled by default but I'll also let @thatch45 and others comment here.

Reply:
My reasoning went along the lines that hitting the disk (I believe you are referring to `salt.utils.minion.running` here) is certainly less expensive than spawning a process. At least that is what I empirically observed in my setup - going above ~100 concurrent processes becomes intractable quickly.

That said I was also wondering whether it would be wiser to disable by default (retaining the current behavior) or implement the safety net by default, which is ultimately what I chose. In any case I am absolutely open to change course if you think it's more appropriate.

Thanks for looking into this @cachedout!

Reply:
My concern is just that we incur that overhead even in cases where we'll never be concerned about hitting an upper limit on processes. We go to great pains to avoid that. The only way I think I'll feel comfortable here is if this is disabled by default, I think.